build-info-extractor/src/main/java/org/jfrog/build/extractor/clientConfiguration/util/GitUtils.java

```diff
@@ -97,13 +97,28 @@ private static String extractVcsRevision(File dotGit, Log log) throws IOExceptio
 
         // Else, if found ref try getting revision using it
         File pathToRef = new File(dotGit, revisionOrRef.ref);
-
-        // Read HEAD file for ref or revision
-        try (BufferedReader br = new BufferedReader(new FileReader(pathToRef))) {
-            String line;
-            while ((line = br.readLine()) != null) {
-                log.debug("Fetched revision from git config: " + line.trim());
-                return line.trim();
+        if (pathToRef.exists()) {
```

Comment:
I suggest we add a unit test specifically for the *extractVcsRevision*, to test the two possible scenarios.